### PR TITLE
Additions to Category Traversal

### PR DESCRIPTION
**Pull request by MattRice12**

### CODE REVIEW

**Comment by rluck0419** [comment]
the huge amount of "diff not rendered" is concerning - this otherwise looks good 👍

**Comment by kate-walters** [comment]
Agreed - there might be something wonky going on with the images. It looks like the images duplicated in a lot of different sizes. Other than that, 👍

**Comment by MattRice12** [comment]
Git gave the following message: `Sorry, we could not display the entire diff because too many files (359) changed.`
So, I'm assuming it's just not displaying the changes even though they are there. All of those changes are seed photos anyways.

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <div> <%= link_to "Create Category", new_category_path, class: "btn" %> </div>
 
 <% categories.each do |category| %>
-  <p><%= link_to category.name, category %></p>
+  <p><%= link_to category.name + " (#{pluralize(category.children.count + category.notes.count + category.photos.count, 'item')})", category %></p>
```
Comment (jah2488) [inline app/views/categories/index.html.erb]: This could be moved into the application helper

File: app/views/categories/show.html.erb [hunk 2]
```diff
@@ -28,7 +28,11 @@
 <% else %>
   <% category.children.each do |child| %>
     <div class="category-name">
-      <%= link_to child.name + " (#{child.children.count + child.notes.count + child.photos.count} items)", child %>
+      <% if child.children.empty? && category.notes.empty? && category.photos.empty? %>
```
Comment (jah2488) [inline app/views/categories/show.html.erb]: This is a lot of calls to `.empty?`. What are we looking for, that none of those things exist?

`[child.children, category.notes, category.photos].all?(&:empty?)` will also do the same thing you're doing here.

File: app/views/categories/show.html.erb [hunk 3]
```diff
@@ -28,7 +28,11 @@
...
+        <%= link_to child.name + " (No items)", child %>
+      <% else %>
+        <%= link_to child.name + " (#{pluralize(child.children.count + child.notes.count + child.photos.count, 'item')})", child %>
```
Comment (jah2488) [inline app/views/categories/show.html.erb]: Would make a good helper method.

**Comment by jah2488** [comment]
There is a lot of duplication to be found in the search controllers that should really be moved into the models. I'm going to merge this one in, but it really should be done soon.

**Pull request merged**